accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,326 @@
+
+#### Patch apply
+
+A patch apply is scheduled using Salt `state.apply` with the [patch install state](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/packages/patchinstall.sls).
+
+#### Mark a system to automatically install patches
+
+Automatic patch installation is implemented at the Uyuni/SUSE Manager level by automatically apply a patch whenever it is available (this action falls into the above one).
+
+#### Issue any power management action via Cobbler
+
+All the power management actions are scheduled via [Cobbler commands](https://cobbler.github.io/manuals/2.8.0/3/1/3_-_Systems.html).
+
+#### Perform an SP migration
+
+An SP migration is scheduled using Salt `state.apply` with the [distupgrade state](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/distupgrade/init.sls).
+
+#### Package upgrade
+
+A package upgrade is scheduled using Salt `state.apply` with the [pkginstall state](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/packages/pkginstall.sls).
+
+As a special case when issuing a package upgrade, a package installation from API is scheduled using Salt `state.apply` with the [pkginstall state](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/packages/pkginstall.sls).
+While it is not possible to install a non-installed package on the target system from the UI, is it possible to upgrade an already installed package by requesting an installation of it from the API. Example:
+```
+ec2-user@ip-172-31-8-0:~> rpm -qa | grep -i bzip2
+bzip2-1.0.6-29.2.x86_64
+
+spacecmd {SSM:0}> system_installpackage 172.31.8.0 bzip2
+172.31.8.0:
+bzip2-1.0.6-30.8.1.x86_64
+
+Start Time: 20191108T11:05:04
+
+Install these packages [y/N]:
+```
+
+#### Package removal
+
+A package removal is scheduled using Salt `state.apply` with the [pkgremove state](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/packages/pkgremove.sls).
+
+#### Limitations
+
+At the time of writing, Uyuni/SUSE Manager or Salt cannot forbid the package operations based on the target packages (there is no such kind of granularity).
+As a future implementation, Uyuni/SUSE Manager _might_ consider adding another level of granularity: filter actions depending on the argument passed (e.g. forbid install of a package when the package is `kubernetes-kubeadm`).
+
+NOTE: It is also implied that the [forbidden operations on a CaaS Platform node are also described in the Uyuni/SUSE Manager documentation](https://github.com/SUSE/doc-susemanager/pull/882/files).
+
+## Design
+
+### Identifying a CaaS Platform node
+
+To identify a CaaS Platform node it is enough to check if the `patterns-caasp-Node` pattern is installed. In the future, we might think to add a check for all CaaS Platform workloads to be installed as well.
```

Comment:
Does CaaSP provide a product package?

Reply:
The only one that seems to be a product package would be `caasp-config`. Why?

Reply:
This is a package we would like to get rid of. `patterns-caasp-Node` is the one to look for.